tiledb/sm/c_api/tiledb.h

```diff
@@ -3201,6 +3234,468 @@ TILEDB_EXPORT int32_t tiledb_array_encryption_type(
+ * @param partitioner The subarray partitioner.
+ * @param attr_name The name of the attribute to set the budget for.
+ * @param budget_off The result size budget (in bytes) for the
+ *     offsets of this attribute.
+ * @param budget_val The result size budget (in bytes) for the
+ *     values of this attribute.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_result_budget_var(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    const char* attr_name,
+    uint64_t budget_off,
+    uint64_t budget_val);
+
+/**
+ * Sets the memory budget (in bytes) that dictates the maximum amount
+ * of bytes to be read and decompressed for the attributes during
+ * reads.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_set_memory_budget(
+ *     ctx, partitioner, 10000000, 20000000);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The partitioner.
+ * @param budget The memory budget for a fixed-sized attribute or the
+ *     offsets of a var-sized attribute.
+ * @param budget_var The memory budget for a var-sized attribute.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_memory_budget(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    uint64_t budget,
+    uint64_t budget_var);
+
+/**
+ * Retrieves the current partition.
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The subarray partitioner.
+ * @param partition The partition to be retrieved.
+ * @return `TILEDB_OK` for success and `TILEDB_OOM` or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_get_current(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    tiledb_subarray_t** partition);
+
+/**
+ * It advances the partitioner to the next partition. If the remaining
+ * subarray cannot be split further to produce the next partition,
+ * ``unsplittable`` is set to ``true``.
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The subarray partitioner.
+ * @param unsplittable Set to ``true`` if the remaining subarray cannot
+ *     be split to produce the next partition.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_next(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    int* unsplittable);
+
+/**
+ * Checks if the partitioner is done producing partitions.
+ *
+ * @param ctx The TileDB context.
+ * @param partitioner The subarray partitioner.
+ * @param done Sets it to ``true`` if the partitioner is done.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_done(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_partitioner_t* partitioner,
+    int* done);
```

Comment:
Let's make this `int32_t*` to be explicit, and also set the docstring to `@param done Sets this to `1` if the partitioner is done, and to `0` otherwise.`